sqlalchemy_redshift/dialect.py

```diff
@@ -744,63 +747,74 @@ def _get_all_relation_info(self, connection, **kw):
             relations[key] = rel
         return relations
 
+    # We fetch column info an entire schema at a time to improve performance
+    # when reflecting schema for multiple tables at once.
     @reflection.cache
-    def _get_all_column_info(self, connection, **kw):
+    def _get_schema_column_info(
+        self, connection, schema=None, **kw
+    ):
+        where_schema = "WHERE schema = :schema" if schema else ""
```

Comment:
This is giving a syntax error when the integration tests run. It's also not clear to me why these variables are named `where_schema` and `and_schema`.

Reply:
Apparently parameter binding doesn't work when using the context manager, which was a surprise to me. I've changed this to just use format strings, and renamed the variable to `schema_clause`. There is a bit of a hack with the spectrum section because it doesn't have a where clause by default.